WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m

```diff
@@ -111,13 +111,24 @@ - (void)viewDidLoad
     DDLogMethod();
     [super viewDidLoad];
     self.navigationItem.title = NSLocalizedString(@"Settings", @"Title for screen that allows configuration of your blog/site settings.");
-    if (self.blog.account) {
-        self.tableSections = @[@(SiteSettingsSectionGeneral), @(SiteSettingsSectionWriting)];
+    if (self.blog.isAdmin) {
+        if (self.blog.account) {
+            self.tableSections = @[@(SiteSettingsSectionGeneral), @(SiteSettingsSectionWriting)];
+        } else {
+            self.tableSections = @[@(SiteSettingsSectionGeneral), @(SiteSettingsSectionAccount), @(SiteSettingsSectionWriting)];
+        }    
+        if ([self.blog supports:BlogFeatureRemovable]) {
+            self.tableSections = [self.tableSections arrayByAddingObject:@(SiteSettingsSectionRemoveSite)];
+        }
     } else {
-        self.tableSections = @[@(SiteSettingsSectionGeneral), @(SiteSettingsSectionAccount), @(SiteSettingsSectionWriting)];
-    }    
-    if ([self.blog supports:BlogFeatureRemovable]) {
-        self.tableSections = [self.tableSections arrayByAddingObject:@(SiteSettingsSectionRemoveSite)];
+        if (self.blog.account) {
+            self.tableSections = @[@(SiteSettingsSectionGeneral)];
+        } else {
+            self.tableSections = @[@(SiteSettingsSectionGeneral), @(SiteSettingsSectionAccount)];
+        }
+        if ([self.blog supports:BlogFeatureRemovable]) {
+            self.tableSections = [self.tableSections arrayByAddingObject:@(SiteSettingsSectionRemoveSite)];
+        }
```

Comment:
Since both branches of this `if` look roughly the same, and we're adding sections (as / if required), maybe something like this?


```
NSMutableArray *tableSections = [NSMutableArray arrayWithObject:@(SiteSettingsSectionGeneral)];

if (!self.blog.account) {
   [tableSections addObject:@(SiteSettingsSectionAccount)];
}

if (self.blog.isAdmin) {
   [tableSections addObject:@(SiteSettingsSectionWriting)];
}

if ([self.blog supports:BlogFeatureRemovable]) {
    [tableSections addObject:@(SiteSettingsSectionRemoveSite)];
}
```